Userland/Libraries/LibWeb/HTML/HTMLMeterElement.cpp

```diff
@@ -22,4 +26,185 @@ void HTMLMeterElement::initialize(JS::Realm& realm)
     set_prototype(&Bindings::ensure_web_prototype<Bindings::HTMLMeterElementPrototype>(realm, "HTMLMeterElement"));
 }
 
+void HTMLMeterElement::visit_edges(Cell::Visitor& visitor)
+{
+    Base::visit_edges(visitor);
+    visitor.visit(m_meter_value_element);
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#concept-meter-actual
+double HTMLMeterElement::value() const
+{
+    auto maybe_value_string = get_attribute(HTML::AttributeNames::value);
+    if (!maybe_value_string.has_value())
+        return 0;
+    auto maybe_value = parse_floating_point_number(maybe_value_string.value());
+    if (!maybe_value.has_value())
+        return 0;
+    return clamp(maybe_value.value(), min(), max());
+}
+
+WebIDL::ExceptionOr<void> HTMLMeterElement::set_value(double value)
+{
+    TRY(set_attribute(HTML::AttributeNames::value, MUST(String::number(value))));
+    update_meter_value_element();
+    document().invalidate_layout();
+    return {};
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#concept-meter-minimum
+double HTMLMeterElement::min() const
+{
+    auto maybe_min_string = get_attribute(HTML::AttributeNames::min);
+    if (!maybe_min_string.has_value())
+        return 0;
+    auto maybe_min = parse_floating_point_number(maybe_min_string.value());
+    return maybe_min.value_or(0);
+}
+
+WebIDL::ExceptionOr<void> HTMLMeterElement::set_min(double value)
+{
+    TRY(set_attribute(HTML::AttributeNames::min, MUST(String::number(value))));
+    update_meter_value_element();
+    document().invalidate_layout();
+    return {};
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#concept-meter-maximum
+double HTMLMeterElement::max() const
+{
+    auto maybe_max_string = get_attribute(HTML::AttributeNames::max);
+    if (!maybe_max_string.has_value())
+        return 1;
+    auto maybe_max = parse_floating_point_number(maybe_max_string.value());
+    if (!maybe_max.has_value())
+        return 1;
```

Comment:
Similar to my comment on `value()`: 1 should be the "candidate" value which we then ensure is >= min.